src/pudl_rmi/connect_deprish_to_ferc1.py

```diff
@@ -66,762 +191,380 @@ def execute(plant_parts_eia, deprish_eia, ferc1_to_eia, clobber=False):
-        candidate_matches_all = pd.merge(
-            candidate_matches_all,
-            candidate_matches_all.groupby(['record_id_eia_deprish'])
-            .agg({'record_id_ferc1': 'count'})
-            .rename(columns={'record_id_ferc1': 'count_ferc1', })
-            .reset_index(),
-        )
-        return candidate_matches_all
-
-    def get_same_true(self, candidate_matches):
-        """
-        Grab the obvious matches which have the same record_id_eia.
 
-        If an candidation match has the same id from both the depreciation
-        records and ferc1.... then we have found a match.
-
-        Args:
-            candidate_matches (pandas.DataFrame): dataframe of the matches
-                of possible matches between the ferc1 and deprecation.
-
-        Returns:
-            pandas.DataFrame: dataframe of matches that have the same EIA
-                record id
-
-        """
-        return candidate_matches[
-            candidate_matches.record_id_eia_deprish ==
-            candidate_matches.record_id_eia_ferc1]
-
-    def get_matches_at_diff_ownership(self, candidate_matches):
-        """
-        Get matches when the record_id_eia matches except for the ownership.
-
-        The master unit list includes various levels of ownership associated
-        with each record. Some are labeled "owned" (for the utilty's owned
-        portion of the plant part) and some are labeled "total" (for the full
-        plant part).
-
-        The method selects the matches where the potential matches have the
-        same EIA record id expect for the ownership level (owned vs total).
-
-        Note: Is there a cleaner way to do this??
+    @property
+    def allocator_cols_dict(self) -> Dict:
+        """Grab the columns from the metadata which need to be allocated."""
+        return {
+            allocate_col: self.treatments[allocate_col].allocator_cols
+            for allocate_col in self.get_treatment_cols('scale')
+        }
 
-        Args:
-            candidate_matches (pandas.DataFrame): dataframe of the matches
-                of possible matches between the ferc1 and deprecation.
-        """
-        diff_own = (
-            candidate_matches[
-                candidate_matches.record_id_eia_deprish.str.replace(
-                    '(owned|total)', "", regex=True) ==
-                candidate_matches.record_id_eia_ferc1.str.replace(
-                    '(owned|total)', "", regex=True)
-            ]
+    @property
+    def plant_part_pk_cols(self):
+        """Get the primary keys for a plant-part."""
+        return (pudl.analysis.plant_parts_eia.PLANT_PARTS
+                [self.plant_part]['id_cols']
+                + pudl.analysis.plant_parts_eia.IDX_TO_ADD
+                + pudl.analysis.plant_parts_eia.IDX_OWN_TO_ADD)
+
+    def execute(self, df_to_scale: pd.DataFrame, ppl: pd.DataFrame):
+        """
+        Scale a dataframe to a generator-level.
+
+        There are four main steps here:
```

Comment:
Re-format list to work in RST (line break separating the block of bulleted items from header). Define the arguments?

Reply:
I believe I fixed this.